sandpack-react/src/components/CodeEditor/CodeMirror.tsx

```diff
@@ -65,17 +65,7 @@ export type Decorators = Array<{
 interface CodeMirrorProps {
   code: string;
   filePath?: string;
-  fileType?:
-    | "js"
-    | "jsx"
-    | "ts"
-    | "tsx"
-    | "css"
-    | "scss"
-    | "less"
-    | "html"
-    | "vue"
-    | "markdown";
+  fileType?: string;
```

Comment:
I'm wondering if we can extend this key using generics and still keep type-safe, let me know if you need any help

Reply:
I've been wracking my brain trying to make this work, but I'm not sure I'm fully understanding how we can use generics in this case. Would love some guidance if you have a chance!